src/app/components/modal-create-project/modal-create-project.component.ts

```diff
@@ -60,43 +59,52 @@ export class ModalCreateProjectComponent implements OnInit, AfterContentChecked
 
   onSystemSelect(system: any) {
     this.selectedSystem = system;
-    if (system.id.startsWith('project')) {
-      this.projCreateForm.get('linkProject').setValue(true);
-    } else {
-      this.projCreateForm.get('linkProject').setValue(false);
-    }
+    this.projCreateForm.get('watchContent').setValue(false);
   }
 
-  createRapidProject() {
-    this.errorMessage = '';
-    const req: RapidProjectRequest = new RapidProjectRequest(this.rapidFolder.system, this.rapidFolder.path);
-    this.projectsService.createRapidProject(req).subscribe( (project) => {
-      this.close(project);
-    }, (err) => {
-     this.errorMessage = err.toString();
-    });
+  setFileName(ev: any) {
+    this.projCreateForm.get('fileName').setValue(ev.target.value);
   }
 
   submit() {
     this.submitting = true;
     const p = new Project();
+    const pr = new ProjectRequest();
+
     p.description = this.projCreateForm.get('description').value;
     p.name = this.projCreateForm.get('name').value;
-    this.projectsService.create(p).subscribe( (project) => {
-      if (this.projCreateForm.get('exportMapLink').value) {
-        const path = this.selectedFiles.length > 0 ? this.selectedFiles[0].path : this.currentPath;
-        const systemId = this.selectedSystem.id;
-        this.projectsService.exportProject(project,
-                                           systemId,
-                                           path,
-                                           this.projCreateForm.get('linkProject').value,
-                                           this.projCreateForm.get('fileName').value)
-      }
+    p.system_path = this.selectedFiles.length > 0 ? this.selectedFiles[0].path : this.currentPath;
+    p.system_id = this.selectedSystem.id;
+    p.system_file = this.projCreateForm.get('fileName').value
+      ? this.projCreateForm.get('fileName').value
+      : p.name;
+
+    if (this.selectedSystem.id.includes('project')) {
+      pr.observable = true;
+      pr.watch_content = this.projCreateForm.get('watchContent').value;
+    } else {
+      pr.observable = this.projCreateForm.get('watchContent').value;
+      pr.watch_content = pr.observable;
+    }
+
+    this.errorMessage = '';
+
+    pr.project = p;
+
+    this.projectsService.create(pr).subscribe((project) => {
       this.close(project);
     }, err => {
-      this.errorMessage = err.toString();
+      this.errorMessage = err.error.message;
```

Comment:
We chatted about this section here 👍 .  As it doesn't involve staging/prod, we can ignore until a different PR or make more generic in this PR . whatever makes sense but just noting it as I wanted to pass along the info below 👇 

-----

For the geoapi-backend problem of not handling the backend exception and creating our expected response **when doing local dev**, it was unclear how to fix that. I thought i had previously but doesn't seem the case. its related to flask-restplus.  and instead of trying to fix that in the no longer supported `flask-restplus`, I think we need to just migrate to something that is supported. I created a ticket for that: https://jira.tacc.utexas.edu/browse/DES-2154.